File: 240322/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.py
import sys

n = int(sys.stdin.readline())
nums = list(map(int,sys.stdin.readline().split()))

ans = 0

for i in range(n-1):
    for j in range(i+2,n):
        ans = max(ans,nums[i]+nums[j])

print(ans)

File: 240404/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.c
#include <stdio.h>

int main() {
    int N, nums[100], ans = 0, temp;

    scanf("%d",&N);
    for(int i = 0; i < N; i++)
        scanf("%d",&nums[i]);
    
    for(int i = 0; i < N; i++)
        for(int j = i+2; j < N; j++) {
            temp = nums[i] + nums[j];
            if (ans < temp)
                ans = temp;
        }
    
    printf("%d\n",ans);
    return 0;
}